students/jjakub/session04/trigram.py

```diff
@@ -0,0 +1,50 @@
+
+
+# words = "I wish I may I wish I might".split()
+# wordcnt = 25
+
+# create dictionary: loop method
+# d = {}
+# for i in range(len(words)-2):
+#     pair = tuple(words[i:i + 2])
+#     follower = words[i + 2]  
+#     if pair not in d:
+#         d[pair] = [follower]
+#     else:
+#         d[pair].append(follower)
+
+### create dictionary
+def build_trigrams(words, wordcnt):
+    d = {}
+    for i in range(len(words)-2):
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]    
+        d.setdefault(pair,[]).append(follower)
+
+    trigrams = {}
+    import random
+    trigrams = list(random.choice(list(d.keys())))
+
+    while len(trigrams) < wordcnt:
+        cur_pair = trigrams[-2:]
+        if tuple(cur_pair) not in d:
+            cur_pair = list(random.choice(list(d.keys())))
+        trigrams.extend(d.setdefault(tuple(cur_pair),[]))
+
+    return trigrams
+
+if __name__ == "__main__":
+    trigrams = build_trigrams(words, wordcnt)
+    print(trigrams)
+
+
+filepath = '.\\students\\jjakub\\session04\\'
```

Comment:
I'm responding to your comment/remark that it doesn't work in the command line. Yes, that's likely the case, if you run your command line from your `session04` directory, not from the git repo root directory (in which case you'll need to run `python students/jjakub/session04/trigram.py`, which is quite inconvenient). If you'd like to avoid this inconvenience, you may start a new VS Code window, and then open your `students/jjakub/session04` folder, instead of the git repo.